### DISCUSSION TITLE
Advice for a terminal app with header, footer and all app output

### DISCUSSION THREAD

**Discussion by charlbury** [Textualize/textual#4697] [Q&A]
Can you advise on the best approach to modify my terminal apps to use Textual to have a permanent Header and Footer, but then show all app output in the body of the terminal?

I have several permanently running terminal apps that collect and process data. I display the progress of these apps in terminal output as they run. I want to add Textual Header and Footer where I can show a summary of the info like Version number, Number of iterations, Number of Errors, Number of Successes etc. etc.

Does anyone have an example of such a Textual app? What would be the best approach and in particular which Textual widget would I use for the body of the display that shows the ever growing output from the app?

**Comment by TomJGooding** [comment]
I'd recommend checking out the [How To Design a Layout](https://textual.textualize.io/how-to/design-a-layout/) in the Textual docs, which includes a section on how to dock a header and footer.

Have you checked the [widget gallery](https://textual.textualize.io/widget_gallery/) in the docs?
